Add --me option for issue command

This should give the list of the issues where the current user is the assignee

```Haskell
import GitHub.Data.Issues (issueAssignees)
import qualified Data.Vector as V

getUser :: IO Text
getUser = "git" $| ["config", "user.login"]
```
```Haskell
getIssueWithAssignee :: IO ()
getIssueWithAssignee = withOwnerRepo (\t o r -> issuesForRepo' t o r stateOpen) >>= \case
    Left err -> errorMessage $ show err
    Right is -> for_ isWA (putTextLn . showIssueName blueCode)
        where isWA = [issue | issue <- is, getUser >>= (\user -> user `V.elem` issueAssignees)]
```

```bash
/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:42:40: error:
    • Couldn't match expected type ‘[a]’
                  with actual type ‘V.Vector Issue’
    • In the expression: is
      In a stmt of a list comprehension: issue <- is
      In the expression:
        [issue |
           issue <- is, getUser >>= (\ user -> user `V.elem` issueAssignees)]
    • Relevant bindings include
        isWA :: [a] (bound at src/Hit/Issue.hs:42:15)
   |
42 |         where isWA = [issue | issue <- is, getUser >>= (\user -> user `V.elem` issueAssignees)]
   |                                        ^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:42:44: error:
    • Couldn't match expected type ‘Bool’ with actual type ‘IO b0’
    • In the expression:
        getUser >>= (\ user -> user `V.elem` issueAssignees)
      In a stmt of a list comprehension:
        getUser >>= (\ user -> user `V.elem` issueAssignees)
      In the expression:
        [issue |
           issue <- is, getUser >>= (\ user -> user `V.elem` issueAssignees)]
   |
42 |         where isWA = [issue | issue <- is, getUser >>= (\user -> user `V.elem` issueAssignees)]
   |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:42:66: error:
    • Couldn't match expected type ‘IO b0’ with actual type ‘Bool’
    • In the expression: user `V.elem` issueAssignees
      In the second argument of ‘(>>=)’, namely
        ‘(\ user -> user `V.elem` issueAssignees)’
      In the expression:
        getUser >>= (\ user -> user `V.elem` issueAssignees)
   |
42 |         where isWA = [issue | issue <- is, getUser >>= (\user -> user `V.elem` issueAssignees)]
   |                                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:42:80: error:
    • Couldn't match expected type ‘V.Vector Text’
                  with actual type ‘Issue
                                    -> V.Vector GitHub.Data.Definitions.SimpleUser’
    • Probable cause: ‘issueAssignees’ is applied to too few arguments
      In the second argument of ‘V.elem’, namely ‘issueAssignees’
      In the expression: user `V.elem` issueAssignees
      In the second argument of ‘(>>=)’, namely
        ‘(\ user -> user `V.elem` issueAssignees)’
   |
42 |         where isWA = [issue | issue <- is, getUser >>= (\user -> user `V.elem` issueAssignees)]
   |                                                                                ^^^^^^^^^^^^^^


```

These are the erros I am facing. Am I going in right direction ??

```Haskell
-- / Get Issue with current user assignee
getIssueWithAssignee :: IO ()
getIssueWithAssignee = withOwnerRepo (\t o r -> issuesForRepo' t o r stateOpen) >>= \case
    Left err -> errorMessage $ show err
    Right is -> for_ isWA (putTextLn . showIssueName blueCode)
        where isWA = [issue | issue <- toList is, filterIssue issue]

-- / Filter issue with current user as assignee
filterIssue :: Issue -> Bool
filterIssue is = getUser >> \user -> user `V.elem` issueAssignees simpleUserLogin userName
```

```bash
/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:47:18: error:
    • Couldn't match expected type ‘Bool’ with actual type ‘IO b0’
    • In the expression:
        getUser
          >> \ user -> user `V.elem` issueAssignees simpleUserLogin userName
      In an equation for ‘filterIssue’:
          filterIssue is
            = getUser
                >> \ user -> user `V.elem` issueAssignees simpleUserLogin userName
   |
47 | filterIssue is = getUser >> \user -> user `V.elem` issueAssignees simpleUserLogin userName
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:47:29: error:
    • Couldn't match expected type ‘IO b0’
                  with actual type ‘a0 -> Bool’
    • The lambda expression ‘\ user
                               -> user `V.elem` issueAssignees simpleUserLogin userName’
      has one argument,
      but its type ‘IO b0’ has none
      In the second argument of ‘(>>)’, namely
        ‘\ user -> user `V.elem` issueAssignees simpleUserLogin userName’
      In the expression:
        getUser
          >> \ user -> user `V.elem` issueAssignees simpleUserLogin userName
   |
47 | filterIssue is = getUser >> \user -> user `V.elem` issueAssignees simpleUserLogin userName
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:47:52: error:
    • Couldn't match expected type ‘(GitHub.Data.Definitions.User
                                     -> Maybe Text)
                                    -> V.Vector a0’
                  with actual type ‘V.Vector GitHub.Data.Definitions.SimpleUser’
    • The function ‘issueAssignees’ is applied to two arguments,
      but its type ‘Issue -> V.Vector GitHub.Data.Definitions.SimpleUser’
      has only one
      In the second argument of ‘V.elem’, namely
        ‘issueAssignees simpleUserLogin userName’
      In the expression:
        user `V.elem` issueAssignees simpleUserLogin userName
    • Relevant bindings include
        user :: a0 (bound at src/Hit/Issue.hs:47:30)
   |
47 | filterIssue is = getUser >> \user -> user `V.elem` issueAssignees simpleUserLogin userName
   |                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/jigar/demo/haskell-demo/hit-on/src/Hit/Issue.hs:47:67: error:
    • Couldn't match expected type ‘Issue’
                  with actual type ‘GitHub.Data.Definitions.SimpleUser
                                    -> Name GitHub.Data.Definitions.User’
    • Probable cause: ‘simpleUserLogin’ is applied to too few arguments
      In the first argument of ‘issueAssignees’, namely ‘simpleUserLogin’
      In the second argument of ‘V.elem’, namely
        ‘issueAssignees simpleUserLogin userName’
      In the expression:
        user `V.elem` issueAssignees simpleUserLogin userName
   |
47 | filterIssue is = getUser >> \user -> user `V.elem` issueAssignees simpleUserLogin userName
   |                                                                   ^^^^^^^^^^^^^^^


--  While building package hit-on-0.0.0 using:
      /home/jigar/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.3 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1 build lib:hit-on exe:hit --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

I am trying hard but can't seem to figure out how to solve these errors. :sweat_smile:

--- a/hit-on.cabal
+++ b/hit-on.cabal
@@ -40,6 +40,7 @@ library
                      , relude ^>= 0.4
                      , shellmet >= 0.0.0
                      , text
+                     , vector ^>= 0.12
 
   ghc-options:         -Wall
                        -Wincomplete-uni-patterns

--- a/src/Hit/Cli.hs
+++ b/src/Hit/Cli.hs
@@ -14,8 +14,8 @@ import Options.Applicative (Parser, ParserInfo, argument, auto, command, execPar
                             subparser, switch)
 
 import Hit.ColorTerminal (arrow, blueCode, boldCode, redCode, resetCode)
-import Hit.Git (runAmend, runClone, runCommit, runCurrent, runFix, runFresh, runHop, runNew,
-                runPush, runResolve, runSync)
+import Hit.Git (getUsername, runAmend, runClone, runCommit, runCurrent, runFix, runFresh, runHop,
+                runNew, runPush, runResolve, runSync)
 import Hit.Issue (runIssue)
 
 import qualified Data.Text as T
@@ -27,14 +27,16 @@ hit = execParser cliParser >>= \case
     Hop branchName -> runHop branchName
     Fresh branchName -> runFresh branchName
     New issueNum -> runNew issueNum
-    Issue issueNum -> runIssue issueNum
+    Issue issueNum me -> if me
+        then getUsername >>= runIssue issueNum . Just
+        else runIssue issueNum Nothing
     Commit message noIssue -> runCommit message noIssue
     Fix message -> runFix message
     Amend -> runAmend
     Resolve branchName -> runResolve branchName
     Push isForce -> runPush isForce
     Sync -> runSync
-    Current -> runCurrent >>= flip whenJust (runIssue . Just)
+    Current -> runCurrent >>= flip whenJust (flip runIssue Nothing . Just)
     Clone name -> runClone name
 
 ----------------------------------------------------------------------------
@@ -51,7 +53,7 @@ data HitCommand
     = Hop (Maybe Text)
     | Fresh (Maybe Text)
     | New Int
-    | Issue (Maybe Int)
+    | Issue (Maybe Int) Bool
     | Commit Text Bool
     | Fix (Maybe Text)
     | Amend
@@ -87,7 +89,13 @@ newP :: Parser HitCommand
 newP = New <$> issueNumP
 
 issueP :: Parser HitCommand
-issueP = Issue <$> optional issueNumP
+issueP = do
+    num <- optional issueNumP
+    me <- switch
+        $ long "me"
+       <> short 'm'
+       <> help "Assigned to me"
+    pure $ Issue num me
 
 commitP :: Parser HitCommand
 commitP = do

--- a/src/Hit/Git.hs
+++ b/src/Hit/Git.hs
@@ -14,6 +14,8 @@ module Hit.Git
        , runSync
        , runCurrent
        , runClone
+
+       , getUsername
        ) where
 
 import Data.Char (isAlphaNum, isDigit, isSpace)

--- a/src/Hit/Issue.hs
+++ b/src/Hit/Issue.hs
@@ -8,8 +8,9 @@ module Hit.Issue
        , parseOwnerRepo
        ) where
 
+import Data.Vector (Vector)
 import GitHub (Error (..), Id, Issue (..), IssueLabel (..), IssueState (..), Name, Owner, Repo,
-               SimpleUser (..), getUrl, mkId, mkName, untagName)
+               SimpleUser (..), User, getUrl, mkId, mkName, untagName)
 import GitHub.Auth (Auth (OAuth))
 import GitHub.Data.Options (stateOpen)
 import GitHub.Endpoints.Issues (issue', issuesForRepo')
@@ -20,19 +21,29 @@ import Hit.ColorTerminal (arrow, blueBg, blueCode, boldCode, errorMessage, green
                           resetCode)
 
 import qualified Data.Text as T
+import qualified Data.Vector as V
 
 
 -- | Run the @issue@ command.
-runIssue :: Maybe Int -> IO ()
-runIssue = \case
+runIssue :: Maybe Int -> Maybe Text -> IO ()
+runIssue issue me = case issue of
     Just num -> getIssue $ mkIssueId num
-    Nothing -> getAllIssues
+    Nothing  -> getAllIssues me
 
 -- | Get the list of the opened issues for the current project.
-getAllIssues :: IO ()
-getAllIssues = withOwnerRepo (\t o r -> issuesForRepo' t o r stateOpen) >>= \case
+getAllIssues :: Maybe Text -> IO ()
+getAllIssues me = withOwnerRepo (\t o r -> issuesForRepo' t o r stateOpen) >>= \case
     Left err -> errorMessage $ show err
-    Right is -> for_ is (putTextLn . showIssueName blueCode)
+    Right is -> for_ (my is) (putTextLn . showIssueName blueCode)
+  where
+    my :: Vector Issue -> Vector Issue
+    my issues = case me of
+        Just (makeName -> username) -> V.filter (assignedTo username . issueAssignees) issues
+        Nothing                     -> issues
+
+    -- Is the username an element of assignees vector?
+    assignedTo :: Name User -> Vector SimpleUser -> Bool
+    assignedTo user = isJust . V.find ((user ==) . simpleUserLogin)
 
 -- | Get the 'Issue' by given issue number.
 getIssue :: Id Issue -> IO ()

